### PR TITLE
feat: #30778 implement core user registry in AgentOS

### PR DESCRIPTION
**Pull request by cetautomatix-coday**
### What this PR delivers

This PR implements the full backend scope of WZ-30778: a persistent user registry, identity resolution per request, and user context injection into agent system prompts.

---

### Backend

**User entity & persistence**
- `User` entity (`userId`, `externalId`, `email`, `firstname`, `lastname`, `bio`) implementing the standard `Entity` contract
- `FilesystemUserRepository` for file-based persistence (consistent with namespace/case persistence strategy)
- `InMemoryUserRepository` for tests

**CRUD API** (`/api/users`)
- Standard endpoints inherited from `EntityController`: `GET /{id}`, `POST /by-ids`, `POST`, `PUT /{id}`, `DELETE /{id}`
- `GET /api/users/me` — resolves the current user from the active security context

**Identity resolution (`SecurityService`)**
- `LocalSecurityService` (default): resolves the OS username, auto-creates the `User` on first access
- `AuthSecurityService`: decodes the Cloudflare JWT / `x-forwarded-email` header, auto-creates the `User` on first access; returns 401 when no identity header is present
- Both implementations use `RequestContextHolder` — no `HttpServletRequest` parameter threading required

**User context injection into agent system prompt**
- `AgentExecutionContext` carries an optional `userId`
- `CaseRuntime.processNextStep` resolves the `userId` from the last `MessageEvent` with `actor.role == USER`
- `CaseServiceImpl.runAgent` validates the user exists before running the agent (case transitions to `ERROR` otherwise)
- `AgentServiceImpl.buildInstructions` appends a `## User` block (id, email, firstname, lastname, bio) to the system prompt when the user resolves — injected in the privileged system-prompt channel so it is never compacted away regardless of conversation length

**OpenAPI spec & TypeScript client regenerated**

---

### Frontend

- User profile page with view/edit modes
- Header avatar linked to the profile page
- Angular client updated from the regenerated OpenAPI spec

---

### What was explicitly deferred

- Case ↔ user ownership link (reverted — tracked in a dedicated ticket)
- SSE streams scoped per user
- Multi-user / delegation routing

---

### Tests

All new behaviour is covered by unit tests following the existing Kotest `StringSpec` + MockK pattern:
- `AgentServiceImplSpec` — namespace + user block injection, all edge cases (null userId, unknown userId, missing optional fields)
- `CaseServiceImplSpec` — user validation gate (null UUID, unknown user → ERROR)
- `CaseRuntimeSpec` — `userId` propagation through the `runAgent` callback
- `UserPersistenceLifecycleSpec` — filesystem round-trip
- `LocalSecurityServiceTest` / `AuthSecurityServiceTest` — identity resolution in both modes

### CODE REVIEW

File: agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/caseFlow/CaseRuntime.kt [hunk 1]
```diff
@@ -282,7 +282,13 @@ class CaseRuntime(
 
                 is AgentRunningEvent -> {
                     logger.info { "[CaseRuntime $id] Found AgentRunningEvent for agent: ${event.agentName}" }
-                    runAgent(event.agentName, eventList.getAll()) { !interruptRequested.get() }
+                    val userId = events
```
Comment (vincent-couturier-whoz) [inline agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/caseFlow/CaseRuntime.kt]: suggestion: could be extracted as a method to clarify the code

File: agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/caseFlow/CaseServiceImpl.kt [hunk 2]
```diff
@@ -115,7 +116,7 @@ class CaseServiceImpl(
             updateStatus = { caseId, newStatus -> handleStatusChange(caseId, newStatus) },
             storeEvent = { event -> storeEvent(event) },
             selectAgent = { content -> selectAgent(content, case.namespaceId, case.id) },
-            runAgent = { agentName, events, shouldContinue -> runAgent(agentName, case.id, events, shouldContinue) },
+            runAgent = { agentName, events, userId, shouldContinue -> runAgent(agentName, case.id, events, userId, shouldContinue) },
```
Comment (vincent-couturier-whoz) [inline agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/caseFlow/CaseServiceImpl.kt]: suggestion: arguments could be named on the call of runAgent

File: agentos/agentos-service/src/test/kotlin/io/whozoss/agentos/agent/AgentServiceImplSpec.kt [hunk 3]
```diff
@@ -179,6 +182,117 @@ class AgentServiceImplSpec : StringSpec() {
...
+                instructions = "You are a helpful assistant.",
+            )
+            val chatClient = mockk<ChatClient>(relaxed = true)
+            every { aiModelRegistry.findByName("my-agent") } returns model
+            every { chatClientProvider.getChatClient("my-agent") } returns chatClient
+            every { userService.findById(userId) } returns user
+
+            val agent = agentService.findAgentByName("my-agent", contextWithUser) as AgentSimple
+
+            agent.instructions!! shouldContain user.email
```
Comment (vincent-couturier-whoz) [inline agentos/agentos-service/src/test/kotlin/io/whozoss/agentos/agent/AgentServiceImplSpec.kt]: suggestion: we should find another name for agent and findAgentByName because it's very confusing with agent definition and case. It seems in the middle but it's not obvious

File: agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/security/AuthSecurityService.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    private fun resolveEmail(cfJwt: String?, forwardedEmail: String?): String? {
+        if (!cfJwt.isNullOrBlank()) {
+            return extractEmailFromJwt(cfJwt)
+        }
+        if (!forwardedEmail.isNullOrBlank()) {
+            logger.info { "[Security/auth] Resolved identity from $X_FORWARDED_EMAIL_HEADER: $forwardedEmail" }
+            return forwardedEmail
+        }
+        return null
```
[the remark this answers is not in the document]
Comment (vincent-audibert-whoz) [reply agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/security/AuthSecurityService.kt]: whened

File: agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/user/UserController.kt
[the remark this answers is not in the document]
Comment (vincent-audibert-whoz) [reply agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/user/UserController.kt]: Test added

File: agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/user/UserController.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ *   GET    /api/users/me     — resolve the caller's own user record via [SecurityService]
+ */
+@RestController
+@RequestMapping(
+    "/api/users",
+    produces = [MediaType.APPLICATION_JSON_VALUE],
+)
+class UserController(
+    private val userService: UserService,
+    private val securityService: SecurityService,
```
[the remark this answers is not in the document]
Comment (vincent-audibert-whoz) [reply agentos/agentos-service/src/main/kotlin/io/whozoss/agentos/user/UserController.kt]: Ok, SecurityService injected in UserService, UserController piping only to UserService.
